php-checks/src/main/java/org/sonar/php/checks/RepeatedComplementOperatorCheck.java

```diff
@@ -25,16 +25,32 @@
 import org.sonar.plugins.php.api.tree.expression.UnaryExpressionTree;
 import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
 
+import java.util.HashMap;
+
 @Rule(key = "S2761")
 public class RepeatedComplementOperatorCheck extends PHPVisitorCheck {
 
-  private static final String MESSAGE = "Use the \"{!|~|-}\" operator just once or not at all. ";
+  private static final String MESSAGE = "Use the \"%s\" operator just once or not at all.";
+  private static final String ADDITIONAL_CAST_MESSAGE = " If a type cast is intended, use \"%s\" instead.";
+
 
   private static final Kind[] COMPLEMENT_UNARY = {
     Kind.BITWISE_COMPLEMENT,
     Kind.LOGICAL_COMPLEMENT,
     Kind.UNARY_MINUS};
 
+  private static final Kind[] CAST_UNARY = {
+    Kind.LOGICAL_COMPLEMENT,
+    Kind.UNARY_MINUS};
+
+  private static final HashMap<Kind, String> CAST_FUNCTION = new HashMap<>();
+
+  public RepeatedComplementOperatorCheck() {
+    CAST_FUNCTION.put(Kind.LOGICAL_COMPLEMENT, "(bool)");
+    CAST_FUNCTION.put(Kind.UNARY_MINUS, "(int)");
+  }
```

Comment:
It's weird to modify a `static final` field inside a constructor.

```suggestion
  private static final Map<Kind, String> CAST_FUNCTION = castFunctions();

  public static Map<Kind, String> castFunctions() {
    Map<Kind, String> map = new HashMap();
    map.put(Kind.LOGICAL_COMPLEMENT, "(bool)");
    map.put(Kind.UNARY_MINUS, "(int)");
    return map;
  }
```